spec/features/diff_viewer_spec.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require "#{Rails.root}/lib/replica"
+
+describe 'diff viewer', type: :feature, js: true do
+  let(:en_wiki) { Wiki.get_or_create(language: 'en', project: 'wikipedia') }
+  let(:course) { create(:course, id: 1, start: '2013-01-01', end: '2014-01-02') }
+  let(:user) { create(:user, id: 1, username: 'I enjoy sandwiches') }
+  let(:article) do
+    create(:article, id: 1, title: 'Periodontium', mw_page_id: 1172296, rating: 'fa')
+  end
+
+  it 'checks whether diff viewer is working properly' do
+    VCR.use_cassette 'diff_viewer/revisions' do
+      login_as user
+      all_users = [
+        build(:user, username: 'I enjoy sandwiches')
+      ]
+      rev_start = 2016_04_17_003430
+      rev_end = 2016_04_18_003430
+
+      response = Replica.new(en_wiki).get_revisions(all_users, rev_start, rev_end)
+
+      response[article.mw_page_id.to_s]['revisions'].uniq.each do |revision|
```

Comment:
It looks like you're basically recreating revisions that would get created via a course update process normally. This is confusing and couples this test far too tightly to `Replica` and the revision importer, which it has nothing to do with.

I suggest either just hard-coding the data you care about so that the test doesn't need to refetch it from Replica each time, or setting up this test so that you can do a UpdateCourseStats and it will pull in the revisions you need based on the dates and the enrolled user.

If you go the latter route, this could be a good starting point for other tests of the more interactive frontend features as well.

Reply:
Did thorugh the `UpdateCourseStats` route !